docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,549 @@
+How to write a training loop in Chainer
+```````````````````````````````````````
+
+.. currentmodule:: chainer
+
+In this tutorial section we will learn how to train a deep neural network to
+classify hand-written digits using the popular MNIST dataset. This dataset
+contains 50000 training examples and 10000 test examples. Each example contains
+a 28x28 greyscale image and a corresponding class label for the digit. Since
+the digits 0-9 are used, there are 10 class labels.
+
+Chainer provides a feature called :class:`~chainer.training.Trainer` that can
+be used to simplify the training process. However, we think it is good for
+first-time users to understand how the training process works before using the
+:class:`~chainer.training.Trainer` feature. Even advanced users might sometimes
+want to write their own training loop and so we will explain how to do so here.
+
+The complete training process consists of the following steps:
+
+1. Prepare a datasets that contain the train/validation/test examples.
+2. Optionally set of iterators for the datasets.
+3. Write a training loop that performs the following operations in each iteration:
+    A. Retreive batches of examples from the training dataset.
+    B. Feed the batches into the model.
+    C. Run the forward pass on the model to compute the loss.
+    D. Run the backward pass on the model to compute the gradients.
+    E. Run the optimizer on the model to update the parameters.
+    F. (Optional): Ocassionally check the performance on a validation/test set.
+
+1. Prepare the dataset
+''''''''''''''''''''''
+
+Chainer contains some built-in functions that can be used to download and
+return Chainer-formatted versions of popular datasets used by the ML and deep
+learning communities. In this example, we will use the built-in function that
```

Comment:
Yeah, the fact that the dataset can be a Python's built-in list should be explained somewhere, but it might be better to be included in the other tutorial: https://github.com/chainer/chainer/pull/2620 rather than in this tutorial that doesn't use Trainer class. How do you think about it?